libsolidity/ast/Types.h

```diff
@@ -725,11 +737,9 @@ class CompositeType: public Type
 class ReferenceType: public CompositeType
 {
 protected:
-	explicit ReferenceType(DataLocation _location): m_location(_location) {}
+	explicit ReferenceType(DataLocation _location) { m_location = _location; }
```

Comment:
Why not add `_location` as a constructor parameter in `CompositeType`?

By the way, the default value of `DataLocation::Storage` was superfluous because the only constructor in `ReferenceType` required this parameter to be specified anyway.

Reply:
Done.